keps/sig-node/4265-proc-mount/README.md

```diff
@@ -0,0 +1,479 @@
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name: `kubelet_pod_start_sli_duration_seconds`
+  - [Optional] Aggregation method:
+  - Components exposing the metric: kubelet
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+- A CRI implementation that supports this feature
+    - All supported versions currently do.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+ProcMountType in the pod spec
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+Negligible
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+Potentially a malicious user given access and running with a root container in the host context could mess with the host processes.
+PSS has already been configured to mitigate this by required a user be in a privileged namespace to get access to the field.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+No effect
+
+###### What are other known failure modes?
+
+Malicious user, protected by PSS.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+The field can be unset in a pod spec to see if SLOs met after the feature is disabled.
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+2018-05-07: k/community update opened
+2018-05-27: k/kubernetes PR merged with support.
+2023-10-02: KEP opened and targeted at Beta
+
+## Drawbacks
+
+
+## Alternatives
```

Comment:
Has anyone tried to submit a patch to the kernel to allow mounting a new procfs with masks?

Reply:
that's a cool idea! I have no idea if it's been tried or how difficult it would be

Reply:
cc: @giuseppe

Reply:
added as an alternative

Reply:
I'm not aware of any effort to allow it.  AFAIK the risk is that allowing a full proc mount will enable a user to access the previously masked files.

This should be safer together with user namespaces, at least it will be possible to allow nested containers in the userns limited environment.

Tagging @brauner in case he has better news/ideas from the kernel side :-)

Reply:
The response was no in 2018: https://patchwork.kernel.org/project/linux-fsdevel/patch/20180404115311.725-1-alban@kinvolk.io/ -- I don't think much has changed really.

I do believe something similar to the LXD approach (I mention below) could be an idea with the advantage it can apply to /sys too.